bindings/android/libparsec/rust/Cargo.toml

```diff
@@ -8,8 +8,8 @@ crate-type = ["staticlib", "cdylib"]
 
 [dependencies]
 libparsec = { path = "../../../../libparsec" }
-lazy_static = "1.4.0"
+lazy_static = { workspace = true }
 jni = "0.21.1"
```

Comment:
I think we should also move the version of the dependencies that are needed once in the workspace.

This is because it is likely with time we will endup with this dependency needed at multiple places (see for instance `base64` which seems really common and is only in `libparsec_client_connection` so far)

We want a no brainer when adding a new dependency, so:
- add the dependency in the crate it is needed and you're done (i.e. the current way to do)
- add the dependency in teh crate it is needed, and ensure it is listed in the workspace

However  the way to do with this PR is: add the depedency in the crate, then search in the whole project if the dependency is also used somewhere, if that's the case ensure it is listed in the workspace, else put the version in the crate. This is complex and we can be pretty sure the dev will go the easy way and just put the dependency with version in the crate (especially given some other dependency are declared this way)

So I'd say we want to have every version declared in the workspace

Reply:
Done, all dependencies are now define in the root `Cargo.toml`, same goes for our own lib as I find it simpler to handle the path on the root `Cargo.toml`